Signal/src/views/GroupTableViewCell.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import UIKit
+
+@objc class GroupTableViewCell: UITableViewCell {
+
+    let TAG = "[GroupTableViewCell]"
+
+    private let avatarView = AvatarImageView()
+    private let nameLabel = UILabel()
+    private let subtitleLabel = UILabel()
+
+    init() {
+        super.init(style: .default, reuseIdentifier: TAG)
+
+        self.contentView.addSubview(avatarView)
+
+        let textContainer = UIView.container()
+        textContainer.addSubview(nameLabel)
+        textContainer.addSubview(subtitleLabel)
+        self.contentView.addSubview(textContainer)
+
+        // Font config
+        nameLabel.font = UIFont.ows_dynamicTypeBody()
+        subtitleLabel.font = UIFont.ows_footnote()
+        subtitleLabel.textColor = UIColor.ows_darkGray()
+
+        // Listen to notifications...
+        // TODO avatar, group name change, group membership change, group member name change
+
+        // Layout
+
+        nameLabel.autoPinEdgesToSuperviewEdges(with: UIEdgeInsets.zero, excludingEdge: .bottom)
+        subtitleLabel.autoPinEdgesToSuperviewEdges(with: UIEdgeInsets.zero, excludingEdge: .top)
+        subtitleLabel.autoPinEdge(.top, to: .bottom, of: nameLabel)
+
+        avatarView.autoPinLeadingToSuperview()
```

Comment:
Don't we need the swift equivalent of:
```
    self.preservesSuperviewLayoutMargins = YES;
    self.contentView.preservesSuperviewLayoutMargins = YES;
```
in order for the avatar to align properly?

Reply:
I believe this would be true if we were using the stock purelayout: `autoPinEdge(toSuperviewEdge: .leading)`.

However, this is one of our own custom purelayout methods, which as you can see in the screenshot, respects the superview margins. Maybe we should rename these to something like: `avatarView.autoPinLeadingToSuperviewMargin()`